app/controllers/samples_controller.rb

```diff
@@ -381,6 +381,27 @@ def edit
   # POST /samples
   # POST /samples.json
   def create
+    # Single sample upload path
+    # Check if the client is up-to-date. "web" is always valid whereas the
+    # CLI client should provide a version string to-be-checked against the
+    # minimum version here. Bulk upload from CLI goes to this method.
+    valid_client = false
+    if client_params && client_params[:client]
+      client = client_params[:client]
+      min_version = '0.3.2'
+      if client == "web" || Gem::Version.new(client) >= Gem::Version.new(min_version)
+        valid_client = true
+      end
+    end
+    unless valid_client
+      respond_to do |format|
+        msg = "Outdated command line client. Please run `pip install --upgrade git+https://github.com/chanzuckerberg/idseq-cli.git `"
+        format.json { render json: { status: msg }, status: :upgrade_required }
+        format.html { render json: { status: msg }, status: :upgrade_required }
+      end
+      return
+    end
```

Comment:
I think you can accomplish this with less code.
1. add :client in the permit list of `sample_params`
2. move `params = sample_params` to the top of `create`
3. change your block to:

```
    client = params[:client]
    unless client == "web" || Gem::Version.new(client) >= Gem::Version.new('0.3.2')
        render json: {
          message: "Outdated command line client. Please run `pip install --upgrade git+https://github.com/chanzuckerberg/idseq-cli.git `",
          status: :upgrade_required
        }
      return
    end
```